tests/unit/test-cmd/test.run.js

```diff
@@ -477,6 +479,28 @@ describe('run', () => {
         });
     });
 
+    it('pauses the web-ext process (CTRL+Z in shell console)', () => {
+      const {reloadStrategy, kill} = prepare();
+      const fakeStdin = new tty.ReadStream();
+      const setRawMode = sinon.spy(fakeStdin, 'setRawMode');
+
+      return reloadStrategy({}, {stdin: fakeStdin})
+        .then(() => {
+          fakeStdin.emit('keypress', 'z', {name: 'z', ctrl: true});
+        }).then(() => {
+          assert.ok(kill.called);
+          assert.deepEqual(kill.firstCall.args, [process.pid, 'SIGTSTP']);
+          sinon.assert.callOrder(setRawMode, setRawMode, kill, setRawMode);
+          sinon.assert.calledThrice(setRawMode);
+          assert.equal(setRawMode.firstCall.args[0], true);
+          assert.equal(setRawMode.secondCall.args[0], false);
+          assert.equal(setRawMode.lastCall.args[0], true);
+        })
+        .then(() => {
+          fakeStdin.emit('keypress', 'c', {name: 'c', ctrl: true});
```

Comment:
@saintsebastian Thanks!

I've been checking if there are other tests that can leak this file loop and it looks like there is one more, the fakeStdin related to the "can reload when user presses R in shell console" test (at line 454 of this file).

Another scenario that we should cover is when a test fails to reach the `then` resolve callback because of an unexpected rejection in the promise chains (we should probably pack the `fakeStdin.emit(...)` into a test helper and use the helper to handle both resolve and reject in the end of the Promise chain, something like:
```js
function getDoEmitCtrlC(stdin) {
  return () => stdin.emit('keypress', 'c', {name: 'c', ctrl: true});
}

.then(getDoEmitCtrlC(fakeStdin), getDoEmitCtrlC(fakeStdin));
```

maybe with a name that looks nicer than `getDoEmitCtrlC` :-)